src/main/groovy/net/vivin/gradle/versioning/VersionUtils.groovy

```diff
@@ -140,7 +144,37 @@ class VersionUtils {
             refresh()
         }
 
-        return versions?.find()
+        if(!versionByTag) {
+            return null
+        } else {
+            def revWalk = new RevWalk(repository)
+            def candidateTags = []
+            def toInvestigateForTags = []
+
+            try {
+                toInvestigateForTags.add(revWalk.parseCommit(repository.resolve(Constants.HEAD)))
+
+                for(RevCommit investigatee = toInvestigateForTags.pop();
```

Comment:
> So for example if I had `while(!toInvestigateForTags.empty())`, in my head I am able to directly translate it to "while we still have tags to investigate".

It's even better. Thanks to Groovy truth you can simply say `while(toInvestigateForTags)`.

You are right about this case. The while indeed looks better and more readable here and checking the already investigated before adding is much better of course also.